src/tools/EpicsHostArch.pl

```diff
@@ -45,15 +118,9 @@ sub HostArch {
         return 'solaris-x86'    if m/^i86pc-solaris/;
         return 'freebsd-x86_64' if m/^x86_64-freebsd/;
         return 'freebsd-x86_64' if m/^amd64-freebsd/;
+        return 'darwin-x86'     if m/^x86(_64)?-darwin/;
+        return 'darwin-aarch64' if m/^(arm64|aarch64)-darwin/;
 
-        my ($kernel, $hostname, $release, $version, $cpu) = uname;
-        if (m/^darwin/) {
-            for ($cpu) {
-                return 'darwin-x86'     if m/^x86_64/;
-                return 'darwin-aarch64' if m/^arm64/;
-            }
-            die "$0: macOS CPU type '$cpu' not recognized\n";
-        }
```

Comment:
This is failing on MacOS, because Perl's `$Config{archname}` doesn't contain CPU information at all on Apple builds:
```
Cox$ perl -V:archname
archname='darwin-2level';
```
The original script only matches the start of the archname on MacOS, then uses `uname` to get the CPU type. You deleted that additional step, breaking the script.

I don't know how this affects the new `-g` option processing.

Reply:
Done!